### PR TITLE
Update golang example

### PR DESCRIPTION
**Pull request by kamyuentse**
The golang example is broken, this patch make it work again.

### CODE REVIEW

**Comment by bjorn3** [comment]
Can you please revert the space -> tab changes? They make reviewing harder and are inconsistent with the rest of the file.

**Comment by kamyuentse** [comment]
@bjorn3 done

File: docs/lang-go.md [hunk 1]
```diff
@@ -35,10 +35,11 @@ import (
 )
 
 func main() {
-    store := wasmtime.NewStore(wasmtime.NewEngine())
-    module, err := wasmtime.NewModuleFromFile(store, "gcd.wat")
+    engine := wasmtime.NewEngine()
+    module, err := wasmtime.NewModuleFromFile(engine, "gcd.wat")
     check(err)
-    instance, err := wasmtime.NewInstance(module, []*wasmtime.Extern{})
+    store := wasmtime.NewStore(engine)
```
Comment (bjorn3) [inline docs/lang-go.md:41]: I think it would make sense to move this just after `engine := wasmtime.NewEngine()`.

Comment (kamyuentse) [reply docs/lang-go.md]: Although I have moved this line, I can not reason about, can you share your point?

Comment (bjorn3) [reply docs/lang-go.md]: I can't really. It is more of a "it feels better" than "I have a concrete reason it is better".

**Comment by alexcrichton** [comment]
Thanks for this @kamyuentse!

**Pull request merged**